modules/lib/lib-node/src/main/resources/lib/xp/node.ts

```diff
@@ -399,35 +399,43 @@ export interface Explanation {
     details: Explanation[];
 }
 
+export interface NodeQueryResultHit {
+    id: string;
+    score: number;
+    explanation?: Explanation;
+    highlight?: HighlightResult;
+}
+
+export interface NodeQueryResultSuggestion {
+    [suggestionName: string]: {
+        text: string;
+        length: number;
+        offset: number;
+        options: {
+            text: string;
+            score: number;
+            freq?: number; // only for term
+        }[];
+    }[];
+}
+
 export interface NodeQueryResult {
     total: number;
     count: number;
-    hits: {
-        id: string;
-        score: number;
-        explanation?: Explanation;
-        highlight?: HighlightResult;
-    }[];
+    hits: NodeQueryResultHit[];
     aggregations?: AggregationsResult;
-    suggestions?: {
-        [suggestionName: string]: {
-            text: string;
-            length: number;
-            offset: number;
-            options: {
-                text: string;
-                score: number;
-                freq?: number; // only for term
-            }[];
-        }[];
-    }[];
+    suggestions?: NodeQueryResultSuggestion[];
 }
 
-export type NodeMultiRepoQueryResult = NodeQueryResult & {
-    hits: {
+export type NodeMultiRepoQueryResult = {
```

Comment:
Maybe will be better to declare NodeMultiRepoQueryResult as interface instead of type here?

Reply:
That's up to ye, I didn't change it :)

Reply:
Yes, let's change it to the `interface` here, since it is no longer a union type.

Reply:
Fixed, I assume @rymsha will do the merging?